mllib/src/main/scala/org/apache/spark/ml/classification/RandomForestClassifier.scala

```diff
@@ -126,20 +127,22 @@ class RandomForestClassifier @Since("1.4.0") (
         s" numClasses=$numClasses, but thresholds has length ${$(thresholds).length}")
     }
 
-    val oldDataset: RDD[LabeledPoint] = extractLabeledPoints(dataset, numClasses)
+    val instances = dataset.select(col($(labelCol)).cast(DoubleType), col($(featuresCol))).rdd.map {
+      case Row(label: Double, features: Vector) => Instance(label, 1.0, features)
```

Comment:
We have to pass in `RDD[Instance]` to `RandomForest.run`. I changed this back to use `extractLabeledPoints`